kinto/core/storage/postgresql/__init__.py

```diff
@@ -443,29 +431,24 @@ def delete_all(self, collection_id, parent_id, filters=None,
                    auth=None):
         if with_deleted:
             query = """
-            WITH deleted_records AS (
-                WITH matching_records AS (
-                    SELECT id, parent_id, collection_id
-                        FROM records
-                        WHERE {parent_id_filter}
-                              {collection_id_filter}
-                              {conditions_filter}
-                              {pagination_rules}
-                        {sorting}
-                        LIMIT :pagination_limit
-                )
-                DELETE
-                FROM records
-                USING matching_records
-                WHERE records.id = matching_records.id
-                  AND records.parent_id = matching_records.parent_id
-                  AND records.collection_id = matching_records.collection_id
-                RETURNING records.id, records.parent_id, records.collection_id
+            WITH matching_records AS (
+                SELECT id, parent_id, collection_id
+                    FROM records
+                    WHERE {parent_id_filter}
+                          {collection_id_filter}
+                          AND deleted = FALSE
+                          {conditions_filter}
+                          {pagination_rules}
+                    {sorting}
+                    LIMIT :pagination_limit
```

Comment:
I think we should consider putting a `FOR UPDATE` here, although I don't really have a strong justification for why. It seems like two different queries could select the same not-yet-deleted rows and delete them simultaneously, although they might want to each delete 10. (But this might be a bigger topic than we want to address in this PR; see [this SO discussion](https://dba.stackexchange.com/questions/69471/postgres-update-limit-1)).